controller/tap/apiserver.go

```diff
@@ -87,32 +87,35 @@ func NewAPIServer(
 // ServeHTTP handles all routes for the APIServer.
 func (a *apiServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	a.log.Debugf("ServeHTTP(): %+v", req)
+	err := a.validate(req)
+	if err != nil {
+		a.log.Debug(err)
+		renderJSONError(w, err, http.StatusBadRequest)
+	} else {
+		a.router.ServeHTTP(w, req)
+	}
```

Comment:
Dont you think since you have only an error you can use a more concise construct such as: 
```
	if err := a.validate(req); err != nil {
		a.log.Debug(err)
		renderJSONError(w, err, http.StatusBadRequest)
	}
	a.router.ServeHTTP(w, req)
```

Reply:
Updated.